apps/client/app/routes/_layout.emails.tsx

```diff
@@ -0,0 +1,114 @@
+import { Form, useLoaderData, useNavigate, useNavigation, useSubmit } from "@remix-run/react";
+import { json, LoaderFunctionArgs } from "@remix-run/node";
+import { useEffect } from "react";
+import { MOCKED_EMAILS } from "../../mocks/emails";
+import { Email } from "../../models/email.model";
+
+
+async function getEmails(searchString: string | null): Promise<Email[]> {
+  await new Promise((resolve) => setTimeout(resolve, 500));
+  if (searchString?.length) {
+    return MOCKED_EMAILS.filter((item) => item.title.toLowerCase().includes(searchString.toLowerCase()));
+  }
+  return MOCKED_EMAILS;
+}
+
+export const loader = async ({
+                               request,
+                             }: LoaderFunctionArgs) => {
+  const url = new URL(request.url);
+  const searchedStr = url.searchParams.get("email");
+  const emails = await getEmails(searchedStr);
+  return json({ emails, searchedStr });
+};
+
+
+const Emails = () => {
+  const { emails, searchedStr } = useLoaderData<typeof loader>();
+  const navigate = useNavigate();
+  const navigation = useNavigation();
+  const submit = useSubmit();
+
+
+  const drawEmailRows = (emails: Email[]) => {
+    const options: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit'
+    };
+
+    return emails.map((email) => (
+      <tr
+        key={ email.id }
+        className="pb-2 border-b-2 border-gray-200 hover:bg-gray-100 cursor-pointer"
+        onClick={() => navigate(`/emails/${ email.id }`)}
+      >
+        <td className="text-sm px-3 py-2">{ email.title }</td>
+        <td className="text-sm px-3 py-1">{ email.status }</td>
+        <td className="text-sm px-3 py-1">{ new Date(email.sentAt).toLocaleDateString("en-US", options) } </td>
+      </tr>
+    ))
+  }
+
+  const searching =
+    navigation.location &&
+    new URLSearchParams(navigation.location.search).has(
+      "email"
+    );
+
+  useEffect(() => {
+    const searchField = document.getElementById("emails-search-input");
+    if (searchField instanceof HTMLInputElement) {
+      searchField.value = searchedStr || "";
+    }
```

Comment:
I think you might want to look into refs from React ;)
Normally you use refs instead of targeting elements with JS